internal/sql/catalog/table.go

```diff
@@ -135,14 +135,29 @@ func (c *Catalog) createTable(stmt *ast.CreateTableStmt) error {
 	} else if err == nil {
 		return sqlerr.RelationExists(stmt.Name.Name)
 	}
+
 	tbl := Table{Rel: stmt.Name}
-	for _, col := range stmt.Cols {
-		tbl.Columns = append(tbl.Columns, &Column{
-			Name:      col.Colname,
-			Type:      *col.TypeName,
-			IsNotNull: col.IsNotNull,
-			IsArray:   col.IsArray,
-		})
+
+	if stmt.ReferTable != nil && len(stmt.Cols) != 0 {
+		return errors.New("create table node cannot have both a ReferTable and Cols")
```

Comment:
@kyleconroy Not sure what to do here. Could raise this while parsing instead of here? Should we create a dedicated error type for this?

Reply:
It's better to catch errors like this here instead of parsing. It's fine to just return an one-off error, especially considering error would come from a mistake inside sqlc as opposed to an error by a user.